src/Cairo.jl

```diff
@@ -89,6 +89,33 @@ end
 
 get_stream_callback(T) = cfunction(write_to_stream_callback, Int32, (Ref{T}, Ptr{UInt8}, UInt32))
 
+
+function read_from_stream_callback(s::IO, buf::Ptr{UInt8}, len::UInt32)
+    # wrap the provided buf into a julia Array
+
+    if VERSION < v"0.5.0-rc"
+        b1 = pointer_to_array(buf,len); #pre-v0.5
+    else
+        b1 = unsafe_wrap(Array{UInt8,1},buf,len);
+    end
+    # read from stream
+    nb = readbytes!(s,b1,len)
+
+    # provide a return status
+    if nb == len
+        r = @compat(Int32(0)) # CAIRO_STATUS_SUCCESS
```

Comment:
agreed. I was distracted, because the write_to_stream callbacks also use hardcoded values.